### PR TITLE
8332610: Remove unused nWakeups in ObjectMonitor

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi,

This comment only serves to confuse people, as `nWakeups` is not shared between threads and never read. It is probably meant for debugging (not sure). Let's delete the comment.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332610](https://bugs.openjdk.org/browse/JDK-8332610): Remove unused nWakeups in ObjectMonitor (**Enhancement** - P4)

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to [f0de0bf1](https://git.openjdk.org/jdk/pull/19326/files/f0de0bf1986186a22d4119ac47a6fe7847393fe5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19326/head:pull/19326` \
`$ git checkout pull/19326`

Update a local copy of the PR: \
`$ git checkout pull/19326` \
`$ git pull https://git.openjdk.org/jdk.git pull/19326/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19326`

View PR using the GUI difftool: \
`$ git pr show -t 19326`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19326.diff">https://git.openjdk.org/jdk/pull/19326.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19326#issuecomment-2122621805)

### CODE REVIEW

**Review by coleenp — CHANGES_REQUESTED** [review]
Please remove that unused variable also in the two places it's defined.  There's a perf counter that does not use this variable so the comment above with TryLock does make sense wrt to that perf counter.  This local variable is unused.

**Comment by jdksjolen** [comment]
> Please remove that unused variable also in the two places it's defined. There's a perf counter that does not use this variable so the comment above with TryLock does make sense wrt to that perf counter. This local variable is unused.

`nWakeups` deleted!

**Review by coleenp — APPROVED** [review]
Looks great! Thank you for finding this unused code.

**Review by dholmes-ora — APPROVED** [review]
Interestingly this code has been dead since it was introduced. I will assume it was leftover debugging/performance-monitoring code.

File: src/hotspot/share/runtime/objectMonitor.cpp [hunk 1]
```diff
@@ -1011,11 +1008,6 @@ void ObjectMonitor::ReenterI(JavaThread* current, ObjectWaiter* currentNode) {
     }
 
     // The lock is still contested.
-    // Keep a tally of the # of futile wakeups.
-    // Note that the counter is not protected by a lock or updated by atomics.
-    // That is by design - we trade "lossy" counters which are exposed to
-    // races during updates for a lower probe effect.
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/objectMonitor.cpp:1017]: The same comment block needs deleting at line 874 too.

Comment (dholmes-ora) [reply src/hotspot/share/runtime/objectMonitor.cpp:1017]: Actually, I think the comment blocks are referring to the `OM_PERFDATA_OP(FutileWakeups, inc());` calls not the `nWakeups` local variable.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Please restore comment, and possibly move them closer to the code the relate to.

Also nit: please change the bug synopsis to "Remove unused nWakeups in ObjectMonitor"

Thanks

**Comment by jdksjolen** [comment]
> Please restore comment, and possibly move them closer to the code the relate to.
> 
> Also nit: please change the bug synopsis to "Remove unused nWakeups in ObjectMonitor"
> 
> Thanks

Good catch on both! I've fixed the issues, thanks.

**Review by coleenp — APPROVED** [review]
I would not have missed that comment, but it looks fine.

**Comment by jdksjolen** [comment]
Errors for maxos-x64 unrelated to this patch. Thanks Coleen, Dan and David.

/integrate

**Review by dcubed-ojdk — COMMENTED** [review]
I'm fine with the final version also.

**Pull request closed without merging**